tools/joffreBulkBooking.js

```diff
@@ -0,0 +1,139 @@
+};
+
+// todo: need to confirm correct values for parkName, facilityName, email, firstName, lastName
+const parkName = 'Joffre Lakes Provincial Park';
+const facilityName = 'Day-Use Trails';
+const email = 'parkinfo@gov.bc.ca';
+const firstName = 'Bulk';
+const lastName = 'Pre-booking';
+const facilityType = 'Trail';
+const type = 'DAY';
+const passStatus = 'reserved';
+
+let passDate, shortPassDate, numberOfGuests, registrationNumber;
+
+const dynamodb = new AWS.DynamoDB(options);
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+
+console.log('\nJOFFRE LAKES BULK BOOKING TOOL\n');
+console.log('Warning: This tool writes directly to the AWS database with very minimal input validation.');
+console.log('Please ensure your inputs are valid and read JOFFRE-README.md before using this tool.');
+console.log('Press CTRL-C to exit.\n');
+console.log(`Park name: ${parkName}`);
+console.log(`Facility name: ${facilityName}`);
+console.log(`Pass type: ${type}`);
+
+rl.question('Booking date (YYYY-MM-DD): ', async function (dateInput) {
+  shortPassDate = dateInput;
+
+  // note: the date is being set to 12 noon UTC time to match the Angular FE app.
+  // Any updates to how dates are recorded in the Angular app should also be made here!
+  passDate = new Date(Date.parse(shortPassDate) + 12 * 60 * 60 * 1000);
+
+  // make the registration number non-random so the sk on the pass is repeatable
+  // and we can overwrite records in the db with a put
+  registrationNumber = '00' + shortPassDate.replace(/-/g, '');
+
+  rl.question('Number of guests?: ', async function (guestsInput) {
+    numberOfGuests = guestsInput;
+
+    // find the booking window for the facility in the db
+    const bookingDaysAhead = +(await getBookingDaysAhead());
+
+    if (isNaN(bookingDaysAhead)) {
+      console.log('bookingDaysAhead for the facility is invalid');
+      process.exit(0);
+    }
+
+    if (passDate - new Date() < (bookingDaysAhead + 1) * 24 * 60 * 60 * 1000) {
+      // check if the booking windows has alread opened.  Add 1 day just to be extra
+      // safe with regard to server and local timezones.
+      console.log(
+        `The ${bookingDaysAhead}-day booking window for this facility/date is already opened or too close to opening.`
+      );
+    } else if (
+      isNaN(numberOfGuests) ||
+      shortPassDate.length !== 10 ||
+      new Date(shortPassDate).getFullYear() !== new Date().getFullYear()
+    ) {
+      console.log('Invalid inputs');
+    } else {
+      const transactItems = { TransactItems: [{ Put: passObj() }, { Put: resCountObj() }] };
+      console.log(transactItems);
+      const response = await dynamodb.transactWriteItems(transactItems).promise();
+      console.log('response:', AWS.DynamoDB.Converter.unmarshall(response));
+      console.log(
+        `Successfully booked ${numberOfGuests} ${type} passes for ${parkName}::${facilityName} on ${shortPassDate}`
```

Comment:
Capacities affected by bulk passes are not being reflected in the front-end admin system.

Reply:
I believe this was happening because this PR is dependent on BRS-309.  I merged the BRS-309 change into this PR so the dependencies are met.

Reply:
We should ensure BRS-309 is fully approved if it is a dependency of this PR. If further changes to BRS-309 are requested, code history may diverge and alter the dependent portions of this PR. Will shift focus to reviewing BRS-309 for now.